# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/test/mochitest/helper_fission_scroll_handoff.html
# Commit: 2e296c018c3b
# Full Hash: 2e296c018c3b62984150b42cec757af680d18695
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2021-02-02 03:35:00
# Regressor Bug: 1687067
# File Overlap Count: 1
# Description:
#   Bug 1687067 - Build OverscrollHandoffChain by walking up hit testing tree instead of APZC tree. r=botond
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D103601
# ==============================================================================

diff -r a32d58612fb7 -r 2e296c018c3b gfx/layers/apz/test/mochitest/helper_fission_scroll_handoff.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/gfx/layers/apz/test/mochitest/helper_fission_scroll_handoff.html	Mon Feb 01 23:25:17 2021 +0000
@@ -0,0 +1,50 @@
+<!DOCTYPE HTML>
+<html>
+<head>
+  <meta charset="utf-8">
+  <title>scroll handoff</title>
+  <script src="/tests/SimpleTest/SimpleTest.js"></script>
+  <script src="/tests/SimpleTest/paint_listener.js"></script>
+  <script src="helper_fission_utils.js"></script>
+  <script src="apz_test_utils.js"></script>
+  <script src="apz_test_native_event_utils.js"></script>
+  <script>
+
+fission_subtest_init();
+
+FissionTestHelper.startTestPromise
+  .then(waitUntilApzStable)
+  .then(loadOOPIFrame("testframe", "helper_fission_empty.html"))
+  .then(waitUntilApzStable)
+  .then(test)
+  .then(FissionTestHelper.subtestDone, FissionTestHelper.subtestFailed);
+
+async function test() {
+  let scrollEventPromise = new Promise(resolve => {
+    window.addEventListener("scroll", resolve, { once: true });
+  });
+
+  let iframe = document.getElementById("testframe");
+
+  synthesizeNativeWheel(iframe, 100, 100, 0, -50);
+  await scrollEventPromise;
+
+  ok(window.scrollY > 0,
+     "Mouse wheel scrolling on OOP iframes in position:fixed subtree " +
+     "should be handed off to the parent");
+}
+
+  </script>
+</head>
+<style>
+iframe {
+    position: fixed;
+    width: 500px;
+    height: 500px;
+}
+</style>
+<body>
+<iframe id="testframe"></iframe>
+<div style="height:1000vh"></div>
+</body>
+</html>
